src/asymmetric_crypto/mod.rs

```diff
@@ -1,9 +1,24 @@
-#[cfg(feature = "curve25519")]
-mod curve_25519;
-#[cfg(feature = "nist_curves")]
-mod nist;
-
-#[cfg(feature = "curve25519")]
-pub use curve_25519::*;
-#[cfg(feature = "nist_curves")]
-pub use nist::*;
+mod curves;
+pub use curves::*;
+
+#[cfg(feature = "rsa")]
+mod rsa;
+#[cfg(feature = "rsa")]
+pub use rsa::*;
+
+pub trait PublicKey {}
+
+pub trait PrivateKey {
+    type PublicKey: PublicKey;
+
+    fn public_key(&self) -> Self::PublicKey;
+}
+
+impl<T: PrivateKey> From<&T> for Box<dyn PublicKey>
+where
+    T::PublicKey: 'static,
+{
+    fn from(private_key: &T) -> Self {
+        Box::new(private_key.public_key())
+    }
+}
```

Comment:
Wouldn't it be better to implement this for each type instead of using a dynamic dispatch?

Reply:
I just left it this for compatibility but I think this should be deprecated. From now on we should use `public_key()` on the private key.
https://github.com/Cosmian/crypto_core/issues/62